source/fundamentals/connection.txt

```diff
@@ -0,0 +1,434 @@
+connect to a MongoDB deployment. It instructs the driver on where and how
+it should connect to MongoDB as well as how it should behave while
+connected. The following example shows each part of the connection URI:
+
+.. figure:: /includes/figures/connection_string_parts.png
+
+In this example, for the protocol, we use ``mongodb+srv`` which specifies the
+:manual:`DNS Seedlist Connection Format
+</reference/connection-string/#dns-seedlist-connection-format>`. This indicates
+that the hostname following it corresponds to the DNS SRV record of your
+MongoDB instance or deployment. If your instance or deployment does not have a
+DNS SRV record, use ``mongodb`` to specify the :manual:`Standard Connection
+Format </reference/connection-string/#standard-connection-string-format>`.
+
+.. note::
+
+   If your deployment is on MongoDB Atlas, follow the
+   :atlas:`Atlas driver connection guide <driver-connection?jmp=docs_driver_java>`
+   to retrieve your connection string.
+
+The next part of the connection string contains your username and password
+if you are using password-based authentication. Replace the value of ``user``
+with your username and ``pass`` with your password. If you are using an
+authentication mechanism that does not require a username and password, omit
+this part of the connection URI.
+
+The next part of the connection string specifies the hostname or IP address and
+port of your MongoDB instance. In the example below, we use ``sample-hostname``
+as the hostname and ``27017`` as the port. Replace these values to point to
+your MongoDB instance.
+
+The last part of the connection string contains connection and authentication
+options as parameters. In the example below, we set two connection options:
+``retryWrites=true`` and ``w=majority``. For more information on connection
+options, skip to the :ref:`connection options <connection-options>` section.
+
+The code below shows how you can use the sample connection URI in a client to
+connect to MongoDB.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/srv.java
+   :language: java
+
+.. _connect-replica-set:
+
+Connect to a Replica Set
+------------------------
+
+A MongoDB replica set deployment is a group of connected instances that
+store the same set of data. This configuration of instances provides data
+redundancy and high data availability.
+
+To connect to a replica set deployment, specify the hostname (or IP address)
+and port numbers of one or more members of the replica set separated by commas.
+Use the :manual:`replica set option
+</reference/connection-string/#urioption.replicaSet>` to specify the name
+of the replica set with which your application is connecting. When the
+replica set option is specified, the driver assumes that your
+connection string uses a seed list of multiple hosts instead of a single
+``host:port`` configuration.
+
+The connection string below specifies three hosts in the cluster and
```

Comment:
Do we need the replicaSet name? The driver doesn't it will work it out from the hosts.  The replicaSet parameter seems like its an old option that serves no real purpose.

Reply:
I'll leave the option/description in the options section (with a note about the fact that it's redundant these days) but I'll remove the bits from "connect to a replica set".